### PR TITLE
AnimatePresence exit fix with fast animations

### PR DESCRIPTION
**Pull request by GianlucaGuarini**
This PR fixes https://github.com/framer/motion/issues/2023

### CODE REVIEW

File: packages/framer-motion/src/components/AnimatePresence/index.tsx [hunk 1]
```diff
@@ -185,19 +185,37 @@ export const AnimatePresence: React.FunctionComponent<
...
+                // compute the keys of children that were rendered once but are no longer present
+                // this could happen in case of too many fast consequent renderings
+                // @link https://github.com/framer/motion/issues/2023
+                const leftOverKeys = Array.from(allChildren.keys()).filter(
+                    (childKey) => !targetKeys.includes(childKey)
+                )
+
+                // clean up the all children map
+                leftOverKeys.forEach((leftOverKey) =>
+                    allChildren.delete(leftOverKey)
```
Comment (GianlucaGuarini) [inline packages/framer-motion/src/components/AnimatePresence/index.tsx:200]: Note that the `allChildren` map was never cleaned up correctly and it was kept growing during all the rendering cycles. This patch fixes this memory leak as well

**Comment by pofixifop** [comment]
I think a lot of people would loooove to have this merged 🥺

**Comment by paavohuhtala** [comment]
I don't want to pressure anyone to do anything, but is something blocking this from being reviewed? We are running into this issue constantly in browser tests (presumably because we use shorter animations in tests to speed things up), and this has been an issue for months.

**Comment by codingwithdidem** [comment]
Any updates? Please we need it to bad

**Comment by yigithanyucedag** [comment]
Any update on this?

**Comment by mattgperry** [comment]
Thanks for the PR and sorry for the delay, I was on paternity leave when this PR was made and it has slipped through the cracks since. I'll merge now and publish tomorrow.

**Review by mattgperry — APPROVED** [review]

**Comment by GianlucaGuarini** [comment]
>  I was on paternity leave when this PR was made...

I know how it feels ;) Take your time and let me know if you need any help on this

**Comment by mattgperry** [comment]
This is out in `10.16.5`

**Pull request merged**